app/models/auth/voms.rb

```diff
@@ -41,15 +41,16 @@ def parse_hash_groups!(hash)
         raise Errors::AuthenticationError, 'voms group env variable is not set' unless hash.key?('GRST_VOMS_FQANS')
         groups = Hash.new { |h, k| h[k] = [] }
         hash['GRST_VOMS_FQANS'].split(';').each do |line|
-          group = parse_group(line)
-          groups.merge!(group) { |_, oldval, newval| oldval + newval } if group
+          group = parse_group!(line)
+          groups.merge!(group) { |_, oldval, newval| (oldval + newval).uniq } if group
```

Comment:
Last comment.

You should avoid calling `uniq` for every iteration (== line). You can easily do this in `roles: value.uniq` below.

Reply:
Yes, good idea. My bad.